### PR TITLE
US152271 - Cleanup folders in the reporter at the beginning of each run

### PR DESCRIPTION
**Pull request by svanherk**
Option 1 - Cleaning up directories before each run in the reporter. Putting up now for comments, still a bit messy. More comments inline.

This includes the new structure we discussed.

### CODE REVIEW

File: src/server/visual-diff-reporter.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	const paths = readdirSync(dir, { withFileTypes: true });
+	paths.forEach(path => {
+		const full = join(dir, path.name);
+		const base = basename(path.name);
+
+		if (path.isDirectory()) {
+			if (base === 'pass' || base === 'fail') _diffPaths.push(full);
+			getDiffPaths(full, _diffPaths);
+		} else {
+			if (base === 'report.json') _diffPaths.push(full);
```
Comment (svanherk) [inline src/server/visual-diff-reporter.js:15]: Put this in as a placeholder, not sure yet if these are individual html files scattered around, one big file at the base, etc.

File: src/server/visual-diff-reporter.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		}
+	});
+	return _diffPaths;
+}
+
+function cleanupFiles(rootDir, testFiles, updateGoldens, runSubset) {
+	const vdiffDir = join(rootDir, '.vdiff');
+	let pathsToClear = [];
+
```
Comment (svanherk) [inline src/server/visual-diff-reporter.js:24]: The logic below is:
- with `--filter` and  `--golden`
  - Delete all directories at the location of the test files to be run (will delete too much if there are sibling and nested test files)
- with `--filter`
  - Delete all `pass` and `fail` directories and `reports` within the location of the test files to be run (will delete too much if there are sibling and nested test files)
- with `--golden`
  - Delete `.vdiff`
- with nothing
  - Delete all `pass` and `fail` directories and `reports` within `.vdiff`

File: src/server/visual-diff-reporter.js (hunk 1 above)
Comment (dlockhart) [reply src/server/visual-diff-reporter.js:15]: Probably will be `report.html`?

Comment (svanherk) [reply src/server/visual-diff-reporter.js:15]: Yeah not sure where I got `json` from lol

**Comment by svanherk** [comment]
Going with https://github.com/BrightspaceUI/testing/pull/36, but can refer back to this if there's issues.

**Pull request closed without merging**